### PR TITLE
Import new versions for packages in xapi-project/xs-opam

### PR DESCRIPTION
**Pull request by psafont**
Not all packages are being updated, but it's a nice chunk. The rest are more difficult to import as they aren't tagged by version.

I expect some changes to be needed before merging, as the quality of the metadata is not as good as the default repo.

### CODE REVIEW

File: packages/cdrom/cdrom.0.9.4/opam [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+version: "0.9.4"
+synopsis: "Query the state and contents of CDROM devices under Linux"
+maintainer: "xen-api@list.xen.org"
+authors: "Dave Scott <dave@recoil.org>"
+tags: "org:xapi-project"
+homepage: "https://github.com/xapi-project/cdrom"
+bug-reports: "https://github.com/xapi-project/cdrom/issues"
+depends: [
+  "ocaml"
+  "dune" {build}
```
Comment (mseri) [inline packages/cdrom/cdrom.0.9.4/opam:12]:
```suggestion
  "dune" {>= "1.4"}
```

File: packages/crc/crc.2.1.0/opam [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+name: "crc"
+authors: [ "David Scott" "John Else" ]
+homepage: "https://github.com/xapi-project/ocaml-crc"
+bug-reports: "https://github.com/xapi-project/ocaml-crc/issues"
+dev-repo: "git+https://github.com/xapi-project/ocaml-crc"
+maintainer: "xen-api@lists.xen.org"
+build: [[ "dune" "build" "-p" name ]]
+run-test: [[ "dune" "runtest" "-p" name "-j" jobs ]]
+depends: [
+  "dune"
```
Comment (mseri) [inline packages/crc/crc.2.1.0/opam:11]:
```suggestion
  "dune" {>= "1.4"}
```

File: packages/dlm/dlm.0.3.1/opam [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+dev-repo: "git+https://github.com/xapi-project/ocaml-dlm.git"
+doc: "https://xapi-project.github.io/ocaml-dlm/doc"
+license: "LGPL-2.1 with OCaml linking exception"
+build: [
+  ["dune" "build" "-p" name "-j" jobs]
+]
+run-test: ["dune" "runtest" "-p" name "-j" jobs]
+depends: [
+  "ocaml"
+  "dune" {build & >= "1.4"}
```
Comment (mseri) [inline packages/dlm/dlm.0.3.1/opam:16]:
```suggestion
  "dune" {>= "1.4"}
```

File: packages/ezxenstore/ezxenstore.0.4.1/opam [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+maintainer: "jonathan.ludlam@citrix.com"
+authors: ["xen-api@lists.xensource.com"]
+license: "ISC"
+homepage: "https://github.com/xapi-project/ezxenstore"
+bug-reports: "https://github.com/xapi-project/ezxenstore/issues"
+dev-repo: "git+https://github.com/xapi-project/ezxenstore.git"
+build: [[ "dune" "build" "-p" name "-j" jobs ]]
+depends: [
+  "ocaml"
+  "dune" {build & >= "1.4"}
```
Comment (mseri) [inline packages/ezxenstore/ezxenstore.0.4.1/opam:11]:
```suggestion
  "dune" {>= "1.4"}
```

File: packages/fd-send-recv/fd-send-recv.2.0.1/opam [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+  "org:xapi-project"
+]
+build: [
+  ["dune" "build" "-p" name "-j" jobs]
+  ["dune" "runtest" "-p" name "-j" jobs] {with-test}
+  ["dune" "build" "@doc" "-p" name] {with-doc}
+]
+depends: [
+  "ocaml"
+  "dune" {build}
```
Comment (mseri) [inline packages/fd-send-recv/fd-send-recv.2.0.1/opam:24]:
```suggestion
  "dune" {>= "1.4"}
```

File: packages/nbd/nbd.4.0.2/opam [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+doc: "https://xapi-project.github.io/nbd/nbd/index.html"
+dev-repo: "git+https://github.com/xapi-project/nbd.git"
+bug-reports: "https://github.com/xapi-project/nbd/issues"
+
+build: ["dune" "build" "-p" name "-j" jobs]
+run-test: ["dune" "runtest" "-p" name "-j" jobs]
+
+depends: [
+  "ocaml" {>= "4.03.0"}
+  "dune" {build}
```
Comment (mseri) [inline packages/nbd/nbd.4.0.2/opam:15]:
```suggestion
  "dune" {>= "1.4"}
```

File: packages/netlink/netlink.0.3.4/opam [hunk 7]
```diff
@@ -0,0 +1,27 @@
+opam-version: "2.0"
+authors: "rob.hoes@citrix.com"
+maintainer: "rob.hoes@citrix.com"
+homepage: "https://github.com/xapi-project/ocaml-netlink"
+bug-reports: "https://github.com/xapi-project/ocaml-netlink/issues"
+dev-repo: "git+https://github.com/xapi-project/ocaml-netlink.git"
+build: ["dune" "build" "-p" name "-j" jobs]
+depends: [
+  "ocaml"
+  "dune" {build}
```
Comment (mseri) [inline packages/netlink/netlink.0.3.4/opam:10]:
```suggestion
  "dune" {>= "1.4"}
```

File: packages/xapi-backtrace/xapi-backtrace.0.7/opam [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+homepage: "https://github.com/xapi-project/backtrace"
+bug-reports: "https://github.com/xapi-project/backtrace/issues"
+dev-repo: "git://github.com/xapi-project/backtrace.git"
+tags: [ "org:xapi-project" ]
+
+build: ["dune" "build" "-p" name "-j" jobs]
+
+depends: [
+  "ocaml"
+  "dune" {build}
```
Comment (mseri) [inline packages/xapi-backtrace/xapi-backtrace.0.7/opam:13]:
```suggestion
  "dune" {>= "1.4"}
```

File: packages/xapi-inventory/xapi-inventory.1.2.1/opam [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+tags: [ "org:xapi-project" ]
+build: [
+  ["./configure" "--default_inventory=%{prefix}%/etc"]
+  ["dune" "build" "-p" name "-j" jobs]
+  ["dune" "runtest" "-p" name "-j" jobs] {with-test}
+]
+depends: [
+  "ocaml"
+  "ocamlfind" {build}
+  "dune" {build}
```
Comment (mseri) [inline packages/xapi-inventory/xapi-inventory.1.2.1/opam:17]:
```suggestion
  "dune" {>= "1.4"}
```

File: packages/xapi-inventory/xapi-inventory.1.2.1/opam [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+dev-repo: "git+http://github.com/xapi-project/xcp-inventory.git"
+tags: [ "org:xapi-project" ]
+build: [
+  ["./configure" "--default_inventory=%{prefix}%/etc"]
+  ["dune" "build" "-p" name "-j" jobs]
+  ["dune" "runtest" "-p" name "-j" jobs] {with-test}
+]
+depends: [
+  "ocaml"
+  "ocamlfind" {build}
```
Comment (mseri) [inline packages/xapi-inventory/xapi-inventory.1.2.1/opam:16]: Is ocamlfind still needed?

File: packages/xapi-rrd/xapi-rrd.1.8.1/opam [hunk 11]
```diff
@@ -0,0 +1,35 @@
...
+tags: [
+  "org:xapi-project"
+]
+build: [
+  ["dune" "build" "-p" name "-j" jobs]
+  ["dune" "runtest" "-p" name] {with-test}
+]
+depends: [
+  "ocaml"
+  "dune" {build & >= "1.4.0"}
```
Comment (mseri) [inline packages/xapi-rrd/xapi-rrd.1.8.1/opam:16]:
```suggestion
  "dune" {>= "1.4"}
```

**Review by mseri — CHANGES_REQUESTED** [review]
Thanks. I only looked briefly, surely we need to fix the omnipresent `build` tag, if anything else is needed (like lower bounds on ocam) will be clearer after the CI runs

File: packages/xapi-inventory/xapi-inventory.1.2.1/opam (hunk 10 above)
Comment (psafont) [reply packages/xapi-inventory/xapi-inventory.1.2.1/opam:16]: It's used for `./configure`

File: packages/xapi-stdext-std/xapi-stdext-std.4.14.0/opam [hunk 12]
```diff
@@ -0,0 +1,26 @@
...
+authors: "xen-api@list.xen.org"
+bug-reports: "https://github.com/xapi-project/stdext/issues"
+dev-repo: "git://github.com/xapi-project/stdext.git"
+homepage: "https://xapi-project.github.io/"
+tags: [ "org:xapi-project" ]
+
+build:  [[ "dune" "build" "-p" name "-j" jobs ]]
+
+depends: [
+  "ocaml"
```
Comment (mseri) [inline packages/xapi-stdext-std/xapi-stdext-std.4.14.0/opam:12]:
```suggestion
  "ocaml" {>= "4.08"}
```
Uses `List.filter_map`

File: packages/xapi-inventory/xapi-inventory.1.2.1/opam [hunk 13]
```diff
@@ -0,0 +1,33 @@
...
+bug-reports: "https://github.com/xapi-project/xcp-inventory/issues"
+dev-repo: "git+http://github.com/xapi-project/xcp-inventory.git"
+tags: [ "org:xapi-project" ]
+build: [
+  ["./configure" "--default_inventory=%{prefix}%/etc"]
+  ["dune" "build" "-p" name "-j" jobs]
+  ["dune" "runtest" "-p" name "-j" jobs] {with-test}
+]
+depends: [
+  "ocaml"
```
Comment (mseri) [inline packages/xapi-inventory/xapi-inventory.1.2.1/opam:15]:
```suggestion
  "ocaml" {>= "4.08"}
```
Uses `List.filter_map`

File: packages/xapi-stdext-unix/xapi-stdext-unix.4.14.0/opam [hunk 14]
```diff
@@ -0,0 +1,29 @@
...
+authors: "xen-api@list.xen.org"
+bug-reports: "https://github.com/xapi-project/stdext/issues"
+dev-repo: "git://github.com/xapi-project/stdext.git"
+homepage: "https://xapi-project.github.io/"
+tags: [ "org:xapi-project" ]
+
+build:  [[ "dune" "build" "-p" name "-j" jobs ]]
+
+depends: [
+  "ocaml"
```
Comment (mseri) [inline packages/xapi-stdext-unix/xapi-stdext-unix.4.14.0/opam:12]:
```suggestion
  "ocaml" {>= "4.08"}
```
Uses `List.filter_map`

File: packages/xapi-rrd/xapi-rrd.1.8.2/opam [hunk 15]
```diff
@@ -0,0 +1,35 @@
...
+dev-repo: "git+https://github.com/xapi-project/xcp-rrd.git"
+tags: [
+  "org:xapi-project"
+]
+build: [
+  ["dune" "build" "-p" name "-j" jobs]
+  ["dune" "runtest" "-p" name] {with-test}
+]
+depends: [
+  "ocaml"
```
Comment (mseri) [inline packages/xapi-rrd/xapi-rrd.1.8.2/opam:15]:
```suggestion
  "ocaml" {>= "4.08"}
```
Uses `List.filter_map` and `Fun`

**Review by mseri — CHANGES_REQUESTED** [review]
A few more changes, the remaining errors seems fine (missing or old C deps, timeouts, ...)

**Review by mseri — CHANGES_REQUESTED** [review]
A few more changes, the remaining errors seems fine (missing or old C deps, timeouts, ...)

**Comment by mseri** [comment]
Looks good now

**Comment by psafont** [comment]
Yeah, was checking if there were some easy fixes...

- Alpine is being weird wrt linux headers
- Centos 8 changed the name for libdlm (dlm-lib)
- rrd code assumes 32-bit ints.

**Comment by mseri** [comment]
I think the only bad one is rrd failure on ppc, everything else looks fine to me

**Comment by avsm** [comment]
Can fix the ppc errors asynchronously, thanks for this!

**Pull request merged**